src/meta_data.rs

```diff
@@ -186,6 +173,74 @@ impl MetaData {
         }
     }
 
+    /// Get node id from request
+    fn get_node_id_from_request(&self, req: &CreateVolumeRequest) -> Option<String> {
```

Comment:
how about merge this part into `select_node`

Reply:
Done